core/node/state_keeper/src/seal_criteria/io_criteria.rs

```diff
@@ -72,7 +76,8 @@ impl IoSealCriteria for TimeoutSealer {
 
     fn should_seal_l2_block(&mut self, manager: &UpdatesManager) -> bool {
         !manager.l2_block.executed_transactions.is_empty()
-            && millis_since(manager.l2_block.timestamp) > self.l2_block_commit_deadline_ms
+            && (millis_since_epoch() - manager.l2_block.timestamp_ms)
```

Comment:
If we 100% sure we don't want to save the precise block timestamp, would it make sense to encapsulate it entirely within state keeper logic? I think all it would take is to make `L2BlockParams` fields crate-private, add a getter for timestamp and a constructor (based on the timestamp in seconds).

Reply:
makes sense